runtime/lua/vim/diagnostic.lua

```diff
@@ -895,22 +895,30 @@ M.handlers.signs = {
     end
 
     local ns = M.get_namespace(namespace)
-    if not ns.user_data.sign_group then
-      ns.user_data.sign_group = string.format('vim.diagnostic.%s', ns.name)
+    if not ns.user_data.sign_ns then
+      ns.user_data.sign_ns = vim.api.nvim_create_namespace('')
     end
 
-    local sign_group = ns.user_data.sign_group
+    local sign_ns = ns.user_data.sign_ns
     for _, diagnostic in ipairs(diagnostics) do
-      vim.fn.sign_place(0, sign_group, sign_highlight_map[diagnostic.severity], bufnr, {
+      local lnum = diagnostic.lnum
+      local col = diagnostic.col
+      local sign_text = opts.sign_text and opts.sign_text[diagnostic.severity] or M.severity[diagnostic.severity]:sub(1,1)
+      local sign_opts = {
+        sign_text = sign_text,
+        sign_hl_group = sign_highlight_map[diagnostic.severity],
         priority = get_priority(diagnostic.severity),
```

Comment:
Note that according to the documentation, `priority` for extmarks only applies to the highlight group, so this may not work as intended here.

Reply:
@lewis6991

Reply:
I think the docs are wrong. The priority should behave exactly as it does for vim-signs.

Reply:
https://github.com/neovim/neovim/blob/6669fc94ae76399a619a350b5d2fc9421e575a81/src/nvim/decoration.c#L386-L392

Reply:
https://github.com/neovim/neovim/issues/19716

Reply:
fixed in #19718